tests/scenes/biosphere/test_leaf_cloud.py

```diff
@@ -24,6 +24,29 @@ def rng():
     yield np.random.default_rng(seed=12345)
 
 
+@pytest.fixture(scope="function")
+def rng_mock():
+    class RngMock:
+        def __init__(self, a, b):
+            self.a = a
+            self.b = b
+
+        def random(self, n=1):
+            returnlist = []
+            for i in range(n):
+                if i % 2:
+                    returnlist.append(self.b)
+                else:
+                    returnlist.append(self.a)
+
+            if len(returnlist) == 1:
+                return returnlist[0]
+            else:
+                return returnlist
+
+    yield RngMock
```

Comment:
How about generalising this with a list?
```suggestion
@pytest.fixture(scope="function")
def rng_mock():
    @attr.s
    class RngMock:
        values = attr.ib(converter=np.atleast_1d)

        def random(self, n=1):
            selector = np.array([i % len(self.values) for i in range(n)])
            return self.values[selector].squeeze()

    yield RngMock
```
This way you can initialise it with any sequence you like.